enumerables.rb

```diff
@@ -0,0 +1,101 @@
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    elements.size.times do
+      yield(elements[index])
+      index += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    elements = is_a?(Range) ? to_a : self
+    index = 0
+    size.times do
+      yield(elements[index], index)
+      index += 1
+    end
+  end
+
+  def my_select(&array)
+    return enum_for unless block_given?
+
+    selected_items = []
+    my_each do |x|
+      selected_items.push(x) unless array.call(x) == false
+    end
+    selected_items
+  end
+
+  def my_all(*args)
+    my_each do |x|
+      return true unless block_given?
+
+      return false unless yield(x)
+    end
+    true
+  end
+
+  def my_any?(*args)
+    args = nil
+    if args
+      my_each { |x| return false unless x == args }
+    end
+    if !block_given?
+      my_each { |x| return true if x }
+    else
+      my_each { |x| return true if x == yield(x) }
+    end
+    false
+  end
+
+  def my_none(*args)
+    !my_any?
+  end
+
+  def my_count(*arguement)
+    counter = 0
+    arguement = nil
+    if arguement
+      my_each { |x| counter += 1 if x == arguement }
+    elsif block_given?
+      my_each { |x| counter += 1 if yield x }
+    else
+      counter = length
+    end
+    counter
+  end
```

Comment:
* [ ] should have working `my_count` identical and returns the same thing as ruby's [`count`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-count). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_count.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_count counts the number of items in enum that are equal to item if an argument is given
```ruby
array = [7,8,9,5,1,0,2,4,0,1]
array.my_count(0) == array.count(0) # false
```